internal/docstore/memdocstore/mem.go

```diff
@@ -151,6 +154,15 @@ func (c *collection) RunActions(ctx context.Context, actions []*driver.Action, o
 		t.Wait()
 	}
 
+	if opts.BeforeDo != nil {
```

Comment:
This doesn't do anything, can we remove it?

Reply:
It lets the user implement a `BeforeDo` that aborts the execution of the list. But yeah, that is probably not useful. But I need it for this test, so I guess we should leave it? If I take it out, how do I get memdocstore to pass the test?